packages/typespec-ts/test/azureIntegration/generated/payload/multipart/src/index.d.ts

```diff
@@ -82,21 +86,27 @@ export declare type FileWithHttpPartOptionalContentTypeRequest = FormData | Arra
 
 export declare interface FileWithHttpPartOptionalContentTypeRequestProfileImagePartDescriptor {
     name: "profileImage";
-    body: HttpPart;
+    body: HttpPartFileOptionalContentType;
+    filename?: string;
+    contentType?: string;
 }
 
 export declare type FileWithHttpPartRequiredContentTypeRequest = FormData | Array<FileWithHttpPartRequiredContentTypeRequestProfileImagePartDescriptor>;
 
 export declare interface FileWithHttpPartRequiredContentTypeRequestProfileImagePartDescriptor {
     name: "profileImage";
-    body: HttpPart;
+    body: HttpPartFileRequiredMetaData;
+    filename?: string;
+    contentType?: string;
 }
 
 export declare type FileWithHttpPartSpecificContentTypeRequest = FormData | Array<FileWithHttpPartSpecificContentTypeRequestProfileImagePartDescriptor>;
 
 export declare interface FileWithHttpPartSpecificContentTypeRequestProfileImagePartDescriptor {
     name: "profileImage";
-    body: HttpPart;
+    body: HttpPartFileSpecificContentType;
+    filename?: string;
+    contentType?: string;
```

Comment:
@v-jiaodi could you check if the newly-added properties are expected?

Reply:
These properties are added when the type is File. So the change is expected. But the relevant condition has issue https://github.com/azure/autorest.typescript/blob/c21658459be2b17cf9ca226004f7c39e12036c4b/packages/rlc-common/src/buildObjectTypes.ts#L115. I think full support in RLC would be covered by https://github.com/Azure/autorest.typescript/issues/2412.

Reply:
why we need to hack in rlc-common for this? I think we should infer whether it's mfd from it's content type not the generated type name?

Reply:
Commented here: https://github.com/Azure/autorest.typescript/issues/2412#issuecomment-2586391463.